A/assimp/build_tarballs.jl

```diff
@@ -0,0 +1,44 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg
+
+name = "assimp"
+version = v"5.0.0"
+
+# Collection of sources required to complete build
+sources = [
+    "https://github.com/assimp/assimp/archive/v5.0.0.tar.gz" =>
+    "b0110a91650d6bb4000e3d5c2185bf77b0ff0a2e7a284bc2c4af81b33988b63c",
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir
+cd assimp-5.0.0/
+mkdir build && cd build
+CMAKE_FLAGS="-DCMAKE_INSTALL_PREFIX=$prefix -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} -DCMAKE_BUILD_TYPE=Release"
+CMAKE_FLAGS="${CMAKE_FLAGS} -DASSIMP_BUILD_ASSIMP_TOOLS=false"
+CMAKE_FLAGS="${CMAKE_FLAGS} -DASSIMP_BUILD_TESTS=false"
+CMAKE_FLAGS="${CMAKE_FLAGS} -DASSIMP_INSTALL_PDB=false"
+cmake ${CMAKE_FLAGS} ..
+make -j${nproc}
+make install
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = supported_platforms()
+
+
+# The products that we will ensure are always built
+products = [
+    LibraryProduct("libassimp", :libassimp)
+]
+
+# Dependencies that must be installed before this package can be built
+dependencies = [
+    PackageSpec(name="Zlib_jll", uuid="83775a58-1f1d-513f-b197-d71354ab007a")
+]
+
+# Build the tarballs, and possibly a `build.jl` as well.
+build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies, preferred_gcc_version=v"7")
```

Comment:
Does this really need gcc v7? Using a late gcc version isn't a big deal for running the library if it's in C, but it can be a problem when you want to link it to another library

Reply:
yes, but only for MinGW:
https://github.com/assimp/assimp/blob/2d2889f73fa1b2ca09ba9f43c9785402d3a7fdd0/CMakeLists.txt#L265

Reply:
I bet we're going to need to `expand_cxxstring_abis()` here.

Reply:
I removed `preferred_gcc_version` and expanded cxxstring ABIs but got more failures.

Reply:
I'm not sure Elliot was suggesting to remove `preferred_gcc_version`, as that one is required by the library, at least for the Windows builds

Reply:
OK, I'll add back. I thought gcc versions and cxxstring ABIs are kinda one-to-one mapped:
> This is used to take, for example, a list of
supported platforms and expand them to include multiple GCC versions for
the purposes of ABI matching.

When gcc is pinned to a specific version, we'll lose some cxxstring ABIs builds, right? Could you elaborate more about when to use `expand_cxxstring_abis`?

Reply:
> I thought gcc versions and cxxstring ABIs are kinda one-to-one mapped

No, they're not: when you use `expand_cxxstring_abis` you duplicate the builds for each compilers: once it builds with `-D_GLIBCXX_USE_CXX11_ABI=1` (for C++11 string ABI), and once with `-D_GLIBCXX_USE_CXX11_ABI=0` (for C++03 string ABI).  More information in [`RootFS.md`](https://github.com/JuliaPackaging/Yggdrasil/blob/dcb1e7c2ff69d456117074b57a27168745296015/RootFS.md#sources-of-incompatibility)

> When gcc is pinned to a specific version, we'll lose some cxxstring ABIs builds, right?

No, see above

> Could you elaborate more about when to use `expand_cxxstring_abis`?

You need it when the C++ string ABI is exposed by the library, so that when loaded against the libstdc++ offered by Julia there must be a matching between the two different ABIs

Reply:
Thanks for the thorough explanation. This also explains some incompatibility issues that I encountered a while ago when trying to build libcxxffi with gcc4. I wish I could read this earlier. 🤦‍♂